embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/api/ApiResponse.kt

```diff
@@ -1,7 +1,21 @@
 package io.embrace.android.embracesdk.comms.api
 
-internal data class ApiResponse<T>(
-    val statusCode: Int?,
-    val headers: Map<String, String>,
-    val body: T?
-)
+/**
+ * ApiResponse is a sealed class that represents the result of an API call.
+ */
+internal sealed class ApiResponse<out T> {
```

Comment:
Do we really need a type here? The body is either a String or it doesn't exist. This seems superfluous?

Reply:
It was already a type in the previous version of ApiResponse, I left it that way in case the body changes at any time to be a stream or something else, so we don't need to make changes in a lot of places. 
But it adds an unnecessary complexity, I'll remove it.

Reply:
Updated